src/base/QueuesClient.ts

```diff
@@ -191,18 +191,37 @@ export default class QueuesClient {
    */
   private getJobs = async <FlowName extends FlowNames>(
     jobIds: string[],
-    resolveChildren: boolean
+    resolveChildren: boolean,
+    keysToGet: (keyof FlowTypes[FlowName]["content"])[] = undefined
   ): Promise<FlowTypes[FlowName]["content"][]> => {
     const transaction = this.redis.multi();
     jobIds.forEach((jobId) => {
       const jobKey = keyFormatter.job(jobId);
-      transaction.hGetAll(jobKey);
+      if (keysToGet) {
+        transaction.hmGet(jobKey, keysToGet as string[]);
+      } else {
+        transaction.hGetAll(jobKey);
+      }
     });
     const rawJobs = await transaction.exec();
-    let jobs: Record<string, any>[] = rawJobs.map((rawJob, index) => ({
-      id: jobIds[index],
-      ...parseObject(rawJob as any, this.logger),
-    }));
+    let jobs: Record<string, any>[] = rawJobs.map((rawJob, index) => {
+      if (!keysToGet) {
+        return {
+          id: jobIds[index],
+          ...parseObject(rawJob as any, this.logger),
+        };
+      }
+
+      const rawJobObject = Object.fromEntries(
+        keysToGet
+          .map<[any, any]>((key, keyIndex) => [key, (rawJob as any)[keyIndex]])
```

Comment:
can we put something better here instead of any,any?

Reply:
I added a string for the first param, and a cast because "Type symbol string | nuimber | symbol is not assignable...", but we pass it to parseObject immediately after, so I think it's probably not worth it to bother with the type here.